common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -94,7 +94,14 @@ public AsciiString(byte[] value, boolean copy) {
      */
     public AsciiString(byte[] value, int start, int length, boolean copy) {
         if (copy) {
-            this.value = Arrays.copyOfRange(value, start, start + length);
+            final byte[] rangedCopy = PlatformDependent.allocateUninitializedArray(length);
+            // DON'T REMOVE: it helps JIT to use a simpler intrinsic stub
```

Comment:
🤦

Reply:
I know I know :( 
i'm now trying different JDK versions to see if I can find a "normal" way to do it and don't get any perf regressions :(

TLDR
copying from char[] ->  byte[] vs byte[] vs byte[] with small sized arrays is usually faster with the former :(
Code on JDK is more modern and better interact with bound checking, while latter is quite old  and never changed in few JDK releases ie System::arrayCopy/Arrays::copyOfRange